### PR TITLE
Rootstrap target 2: Sign Up Feature

### PR DESCRIPTION
**Pull request by juanfopaez**
Sign Up screen: 
<br />

image

### CODE REVIEW

File: src/components/common/SelectField.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  error,
+  id,
+  ...props
+}) => (
+  <div className="selectWrapper">
+    <label htmlFor={id} className={error ? 'error' : ''}>
+      {label}
+    </label>
+    <select className={error ? 'error' : ''} id={id} {...props} {...register} />
+    {error ? <span>{error}</span> : <span />}
```
Comment (eitanlab) [inline src/components/common/SelectField.tsx:24]: Do You need an empty span here? If not just use "&&" instead of "?"

Comment (juanfopaez) [reply src/components/common/SelectField.tsx:24]: I need it, it avoids some resizing and spacing issues (input fields components works in the same way) :)

File: src/pages/SignUp.tsx [hunk 2]
```diff
@@ -1,21 +1,159 @@
...
+              register={register('gender', {
+                required: 'Your gender is required'
+              })}
+              error={errors.gender ? errors.gender.message : ''}
+              defaultValue=""
+            >
+              <option value="" disabled>
+                SELECT YOUR GENDER
+              </option>
+              {Object.values(genders).map((gender) => (
```
Comment (eitanlab) [inline src/pages/SignUp.tsx]: Maybe You can destruct gender here, but is not really important

**Review by eitanlab — APPROVED** [review]

File: src/components/common/SelectField.tsx (hunk 1 above)
Comment (eitanlab) [reply src/components/common/SelectField.tsx:24]: Right!

**Comment by eitanlab** [comment]
Nicely done Juan! I don't have a lot of experience with Jest so I couldn't reviewed it in depth

File: src/pages/SignUp.tsx (hunk 2 above)
Comment (juanfopaez) [reply src/pages/SignUp.tsx]: ✅

File: src/state/reducers/statusReducer.ts [hunk 3]
```diff
@@ -14,10 +14,10 @@ export default createReducer({}, (builder) => {
   builder
     .addMatcher(
       ({ type }) => type.endsWith(`/${actionStatus.REJECTED}`),
-      (state: any, { type, error }) => {
-        state[getActionKey(type)] = {
+      (state: any, action: any) => {
```
Comment (marcio-b) [inline src/state/reducers/statusReducer.ts:17]: Is there any chance of avoiding any here? Do you think you could provide a typing for this action?

File: src/scss/components/common/SelectField.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      outline: none;
+      &.error {
+        border-color: $error;
+      }
+      &:focus {
+        outline-style: none;
+        box-shadow: none;
+      }
+    }
+    span {
```
Comment (marcio-b) [inline src/scss/components/common/SelectField.scss:39]: if span will only be used for errors, it might be a good idea to use a class instead

**Review by marcio-b — COMMENTED** [review]
I posted some comments here even though it is already merged it might help on future PRs... sorry for the delay

**Pull request merged**